### PR TITLE
reading swagger input and adding default security

### PR DESCRIPTION
**Pull request by Shehanir**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: apim-operator/deploy/crds/default_security_cr.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: wso2.com/v1alpha1
+kind: Security
+metadata:
+  name: default-security
+spec:
+  type: JWT
+  certificate: secret-test
+  issuer: https://localhost:9443/oauth2/defaultTest
+  audience: https://localhost:9443/oauth2/defaultTest
```
Comment (praminda) [inline apim-operator/deploy/crds/default_security_cr.yaml:9]: Always end a file with empty new line

File: apim-operator/deploy/crds/wso2_v1alpha1_security_cr.yaml [hunk 2]
```diff
@@ -8,7 +8,4 @@ spec:
   type: Oauth
   certificate: secret-test
   endpoint: https://wso2apim.wso2.svc:9443
-  credentials: secret-test-certificate
-  issuer: https://localhost:9443/oauth2/test
-  alias: my-alias
-  audience: https://localhost:9443/oauth2/test
\ No newline at end of file
+  credentials: secret-test-certificate
```
Comment (praminda) [inline apim-operator/deploy/crds/wso2_v1alpha1_security_cr.yaml:11]: Same as above. Add empty new line

File: apim-operator/pkg/controller/api/api_controller.go [hunk 3]
```diff
@@ -215,9 +215,26 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
 
 	reqLogger.Info("getting security instance")
 
+	//get defined security cr from swagger
+	definedSecurity, checkSecuritykind := swagger.Extensions[securityExtension]
+	var securityName string
+
+	if checkSecuritykind {
+		rawmsg := definedSecurity.(json.RawMessage)
```
Comment (praminda) [inline apim-operator/pkg/controller/api/api_controller.go:223]: Are we using tab to indent `go` src files? For java this is 4 space characters.

File: apim-operator/pkg/controller/api/constants.go [hunk 4]
```diff
@@ -1,25 +1,26 @@
 package api
 
 const (
-	dockerConfig = "docker-config"
+	dockerConfig  = "docker-config"
```
Comment (praminda) [inline apim-operator/pkg/controller/api/constants.go:4]: Why all of the diff in the file is adding unwanted extra spaces? Is this a go formatting standard?
```diff
--dockerConfig  = "docker-config"
++dockerConfig = "docker-config"

File: apim-operator/deploy/crds/wso2_v1alpha1_security_cr.yaml (hunk 2 above)
Comment (Shehanir) [reply apim-operator/deploy/crds/wso2_v1alpha1_security_cr.yaml:11]: File was moved to scenarios and it's fixed in the latest code

File: apim-operator/pkg/controller/api/constants.go (hunk 4 above)
Comment (Shehanir) [reply apim-operator/pkg/controller/api/constants.go:4]: This is how it added with go formatting

File: apim-operator/pkg/controller/api/api_controller.go (hunk 3 above)
Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go:223]: This is added with go fmt (go formatting files) and it uses tabs for indentation.

File: apim-operator/deploy/crds/default_security_cr.yaml (hunk 1 above)
Comment (Shehanir) [reply apim-operator/deploy/crds/default_security_cr.yaml:9]: Added

**Pull request merged**